java_console/ui/src/main/java/com/rusefi/maintenance/ProgramSelector.java

```diff
@@ -166,19 +168,52 @@ private static void flashOpenBltCan(JComponent parent, UpdateOperationCallbacks
         }
     }
 
+    private static Pair<Boolean, String[]> waitForEcuPortDisappeared(
+        final String ecuPort,
+        JComponent parent,
+        final UpdateOperationCallbacks callbacks
+    ) {
+        callbacks.log("Waiting for ECU to reboot to OpenBlt...");
+
+        String[] currentPorts = null;
+        int attemptsCount = 0;
+        for (boolean continueWaiting = true; continueWaiting; attemptsCount++) {
+            // Give the bootloader sec to enumerate
+            BinaryProtocol.sleep(1000);
+
+            currentPorts = LinkManager.getCommPorts();
+            log.info("currentPorts: [" + String.join(",", currentPorts) + "]");
+
+            // Check that the ECU disappeared from the "after" list
+            final boolean ecuPortStillAlive = !PortDetector.AUTO.equals(ecuPort) && Arrays.stream(currentPorts).anyMatch(ecuPort::equals);
+            if (!ecuPortStillAlive) {
+                return new Pair<>(true, currentPorts);
+            } else if (attemptsCount % 3 == 2) {
+                // each 3 seconds we suggest user to terminate waiting:
+                continueWaiting = JOptionPane.YES_OPTION == JOptionPane.showConfirmDialog(
```

Comment:
please no, popup dialogs are evil

I would say show those alive dots as on the red label and give it up to 30 seconds